### PR TITLE
Use `/ui/:repo/:namespace` paths for namespace

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: [AAH-1945](https://issues.redhat.com/browse/AAH-1945)

Filtering by repository doesn't work if user is on `/ui/namespaces/<namespace>` or `/ui/my-namespaces/<namespace>`. We should use `/ui/:repo/:namespace` paths for namespaces instead.

User was redirected to the wrong path after:
- creating namespace
- editing the namespace
- breadcrumbs

- I added a test `Filter by repository` to test correct functionality (could get handy in the future with new repo management features)

### CODE REVIEW

File: src/components/namespace-form/namespace-form.tsx [hunk 1]
```diff
@@ -82,8 +82,9 @@ export class NamespaceForm extends React.Component<IProps> {
                 <Link
                   target='_blank'
                   to={formatPath(
-                    Paths.myCollections,
+                    Paths.namespaceByRepo,
                     {
+                      repo: 'published',
```
Comment (himdel) [inline src/components/namespace-form/namespace-form.tsx:87]: should this alos use `this.context.selectedRepo`?

File: src/containers/namespace-list/namespace-list.tsx [hunk 2]
```diff
@@ -165,8 +165,9 @@ export class NamespaceList extends React.Component<IProps, IState> {
           onCreateSuccess={(result) =>
             this.setState({
               redirect: formatPath(
-                Paths.myCollections,
+                Paths.namespaceByRepo,
                 {
+                  repo: 'published',
```
Comment (himdel) [inline src/containers/namespace-list/namespace-list.tsx:170]: (and here?)

File: src/components/namespace-form/namespace-form.tsx (hunk 1 above)
Comment (jerabekjiri) [reply src/components/namespace-form/namespace-form.tsx:87]: you're right, I thought we could only edit the published repo, but I was wrong.

File: src/containers/namespace-list/namespace-list.tsx (hunk 2 above)
Comment (jerabekjiri) [reply src/containers/namespace-list/namespace-list.tsx:170]: I think after creating a new namespace, user should be redirected to the published repo in UI. I don't think we should redirect to the previous or latest selected repo.

Comment (himdel) [reply src/containers/namespace-list/namespace-list.tsx:170]: Aah, makes sense, works for me :+1: :)

**Review by himdel — APPROVED** [review]
LGTM :+1: :)

(I just restarted tests, there was an imports failure, but looks unrelated .. I think :))

**Pull request merged**